src/coordinator/storage/local/cluster_test.go

```diff
@@ -0,0 +1,135 @@
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package local
+
+import (
+	"testing"
+	"time"
+
+	"github.com/m3db/m3db/src/coordinator/storage"
+	"github.com/m3db/m3db/src/dbnode/client"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func newTestClientFromConfig(ctrl *gomock.Controller) (
+	newClientFromConfig,
+	*client.MockSession,
+) {
+	mockSession := client.NewMockSession(ctrl)
+
+	mockClient := client.NewMockClient(ctrl)
+	mockClient.EXPECT().DefaultSession().Return(mockSession, nil).AnyTimes()
+
+	newClientFn := func(
+		cfg client.Configuration,
+		params client.ConfigurationParameters,
+		custom ...client.CustomOption,
+	) (client.Client, error) {
+		return mockClient, nil
+	}
+
+	return newClientFn, mockSession
+}
+
+func TestNewClustersFromConfig(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	newClient1, mockSession1 := newTestClientFromConfig(ctrl)
+	newClient2, mockSession2 := newTestClientFromConfig(ctrl)
+	cfg := ClustersStaticConfiguration{
+		ClusterStaticConfiguration{
+			newClientFromConfig: newClient1,
+			Namespaces: []ClusterStaticNamespaceConfiguration{
+				ClusterStaticNamespaceConfiguration{
+					Namespace:          "unaggregated",
+					StorageMetricsType: storage.UnaggregatedMetricsType,
+					Retention:          7 * 24 * time.Hour,
+				},
+			},
+		},
+		ClusterStaticConfiguration{
+			newClientFromConfig: newClient2,
+			Namespaces: []ClusterStaticNamespaceConfiguration{
+				ClusterStaticNamespaceConfiguration{
+					Namespace:          "aggregated0",
+					StorageMetricsType: storage.AggregatedMetricsType,
+					Retention:          30 * 24 * time.Hour,
+					Resolution:         time.Minute,
+				},
+				ClusterStaticNamespaceConfiguration{
+					Namespace:          "aggregated1",
+					StorageMetricsType: storage.AggregatedMetricsType,
+					Retention:          365 * 24 * time.Hour,
+					Resolution:         10 * time.Minute,
+				},
+			},
+		},
+	}
+
+	clusters, err := cfg.NewClusters()
+	require.NoError(t, err)
+
+	// Resolve clusters and check attributes
+	unaggregatedNs := clusters.UnaggregatedClusterNamespace()
+	assert.Equal(t, "unaggregated", unaggregatedNs.NamespaceID().String())
+	assert.Equal(t, storage.Attributes{
+		MetricsType: storage.UnaggregatedMetricsType,
+		Retention:   7 * 24 * time.Hour,
+	}, unaggregatedNs.Attributes())
+	assert.True(t, mockSession1 == unaggregatedNs.Session())
+
```

Comment:
Add something like this for the negative case?

```golang
 	_, ok := clusters.AggregatedClusterNamespace(RetentionResolution{
 		Retention:  time.Minute,
 		Resolution: time.Minute,
 	})
 	require.False(t, ok)
```

Reply:
Sure thing.